actions/utils.go

```diff
@@ -46,3 +59,21 @@ func join(A []string, delim string) string {
 
 	return buffer.String()
 }
+
+// Randomly generate a set of indexes in each sector
+func generateInsertedIndexesForPearl(fileSizeInByte int) []int {
+	if fileSizeInByte <= 0 {
+		return nil
+	}
+
+	numOfSectors := int(math.Ceil(float64(fileSizeInByte) / fileSectorInByte))
+	remainderOfChunks := math.Ceil(float64(fileSizeInByte) / fileChunkSizeInByte)
+
+	var indexes []int
```

Comment:
You can make this fixed length
```go
var indexes [numSectors]int
```

Reply:
That won't work, Because numSectors is not a constant at compile time.